### PR TITLE
Add a test for step-70.

### PR DESCRIPTION
**Pull request by bangerth**
#15007 fixes step-70, but does not add a test that verifies that the fix remains viable. This patch does so.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

File: tests/non_matching/step-70.cc [hunk 1]
```diff
@@ -0,0 +1,1983 @@
...
+  MPILogInitAll log(true);
+
+  deallog.depth_console(1);
+  try
+    {
+      std::string prm_file;
+      if (argc > 1)
+        prm_file = argv[1];
+      else
+        prm_file = SOURCE_DIR "step-70.in.prm";
```
Comment (drwells) [inline tests/non_matching/step-70.cc:1928]:
```suggestion
        prm_file = SOURCE_DIR "/step-70.in.prm";
```

**Comment by bangerth** [comment]
Oh, doh! Let's see if this works now.

**Comment by bangerth** [comment]
I don't know what's happening here. On the tester, this test fails with the following output:
```
--------------------------------------------------------
An error occurred in line <79> of file </jenkins/workspace/dealii-mpi_PR-15370/source/base/function_parser.cc> in function
    void dealii::FunctionParser<dim>::initialize(const string&, const std::vector<std::__cxx11::basic_string<char> >&, const std::map<std::__cxx11::basic_string<char>, double>&, bool) [with int dim = 3; std::string = std::__cxx11::basic_string<char>]
The violated condition was: 
    this->n_components == expressions.size()
Additional information: 
    The number of components (4) is not equal to the number of expressions
    (3).
--------------------------------------------------------
```
But locally, this succeeds on my system. I don't really know how to debug this :-(

File: tests/non_matching/step-70.cc [hunk 2]
```diff
@@ -0,0 +1,1983 @@
...
+
+      if (prm_file.find("23") != std::string::npos)
+        {
+          StokesImmersedProblemParameters<2, 3> par;
+          ParameterAcceptor::initialize(prm_file);
+
+          StokesImmersedProblem<2, 3> problem(par);
+          problem.run();
+        }
+      else if (prm_file.find('3') != std::string::npos)
```
Comment (kronbichler) [inline tests/non_matching/step-70.cc:1938]: @bangerth I don't know where this code comes from, but this at least explains the problem: The PR has the number '15370', and the CI MPI tester creates a directory for running this test including the folder name. This then appears in `prm_file` that then triggers the code to run in 3d, as far as I can see. Can you try to remove this brittle part to read off the dimension from the `main()` function, then we can merge this.

I check the test on my machine, where it succeeds.

**Review by kronbichler — APPROVED** [review]

File: tests/non_matching/step-70.cc (hunk 2 above)
Comment (bangerth) [reply tests/non_matching/step-70.cc]: Wow, that's wild!

I've fixed this. Let's see what happens. The code is from the original step-70, as it exists in our examples directory. I'll open an issue for us to fix.

Comment (bangerth) [reply tests/non_matching/step-70.cc]: See #15453.

**Pull request merged**